conn2res/plotting.py

```diff
@@ -0,0 +1,248 @@
+    plt.suptitle(title)
+
+    sns.despine(offset=10, trim=True)
+    if savefig:
+        fig.savefig(fname=os.path.join(FIG_DIR, f'{title}_io.png'),
+                    transparent=True, bbox_inches='tight', dpi=300)
+    plt.show(block=block)
+
+
+def plot_performance_curve(df, title, num=2, figsize=(12, 10), savefig=False, block=True):
+
+    sns.set(style="ticks", font_scale=2.0)
+    fig = plt.figure(num=num, figsize=figsize)
+    ax = plt.subplot(111)
+
+    palette, hue, hue_order = None, None, None
+    if 'module' in df.columns:
+        hue = 'module'
+        n_modules = len(np.unique(df['module']))
+        palette = sns.color_palette('husl', n_modules+1)[:n_modules]
+
+        if 'VIS' in list(np.unique(df['module'])):
+            hue_order = ['VIS', 'SM', 'DA', 'VA', 'LIM', 'FP', 'DMN']
+        else:
+            hue_order = None
+
+    sns.lineplot(data=df, x='alpha', y='score',
+                 hue=hue,
+                 hue_order=hue_order,
+                 palette=palette,
+                 markers=True,
+                 ax=ax)
+
+    sns.despine(offset=10, trim=True)
+    plt.title(title)
+    if savefig:
+        fig.savefig(fname=os.path.join(FIG_DIR, f'{title}_score.png'),
+                    transparent=True, bbox_inches='tight', dpi=300)
+    plt.show(block=block)
+
+
+def plot_time_series(x, feature_set='orig', idx_features=None, n_features=None, xlim=[0, 150], ylim=None,
+                     scaler=1, num=1, figsize=(12, 6), subplot=None, title=None, fname='time_course',
+                     legend_label='', savefig=False, block=True, **kwargs):
+    # transform data
+    x = transform_data(
+        x, feature_set, idx_features=idx_features, n_features=n_features, scaler=scaler, **kwargs)
+
+    # open figure and create subplot
+    plt.figure(num=num, figsize=figsize)
+    if subplot is None:
+        subplot = (1, 1, 1)
+    plt.subplot(*subplot)
+
+    # plot data
+    if x.size > 0:
+        plt.plot(x)
+
+        # add x and y limits
+        plt.xlim(xlim)
+        if ylim is not None:
+            plt.xlim(ylim)
+
+        # plot legend
+        try:
+            legend = [f'{legend_label} {n+1}' for n in range(x.shape[1])]
+        except:
+            legend = [f'{legend_label} 1']
+        try:  # quick fix to get previously plotted legends
+            lg = plt.gca().lines[-1].axes.get_legend()
+            legend = [text.get_text() for text in lg.texts] + legend
+        except:
+            pass
+        if len(legend) <= 5:
+            plt.legend(legend, loc='upper right', fontsize=12)
+        else:
+            plt.legend(legend, loc='upper right', fontsize=12, ncol=2)
+        # plt.legend(legend, loc='upper center', bbox_to_anchor=(
+        #     0.5, 1.05), fontsize=10, ncol=len(legend))
+
+    # set xtick/ythick fontsize
+    plt.xticks(fontsize=22)
+    plt.yticks(fontsize=22)
+
+    # add title
+    if title is not None:
+        plt.title(f'{title} time course', fontsize=22)
```

Comment:
The fontsizes are hard coded at the moment but probably a better solution would be to define these fontsizes (and other rc.Params outside of the function in plotting.py) so that all plotting functions can use the same settings but also allowing to change these settings easily in 1 place.

One solution could be to define a dictionary and update rc.Params but I am not really familiar with all this and I didn't have time to look this up properly. I decided to hard-code simply as when I started using rc.Params I ran into issues with setting xticks and yticks (apparently it is a matplotlib version issue but haven't really looked into it properly).